store/tikv/2pc.go

```diff
@@ -387,6 +421,22 @@ func (c *twoPhaseCommitter) prewriteSingleBatch(bo *Backoffer, batch batchKeys)
 			}
 			log.Debugf("con:%d 2PC prewrite encounters lock: %v", c.connID, lock)
 			locks = append(locks, lock)
+
+			if keyErr.PreconditionErr != nil {
```

Comment:
I think should check precondition error first.

Reply:
Currently, `PreconditionErr` is not treat as an error, the operation is still done even we meet `PreconditionErr`. Under this condition, the check can be put anywhere.